modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/jaxrs/feature/HttpMethodFeature.java

```diff
@@ -155,14 +156,15 @@ private Method _getSuperMethod(Method method) {
 		HttpMethodFeature.class);
 
 	private BundleContext _bundleContext;
+
+	@Reference(target = "(oauth2.scope.checker.type=http.method)")
```

Comment:
I had similar thoughts. In the end I concluded that because other callers (`GraphQLRequestContextValidator`) of the new `ScopeLogic` SPI requires a `ScopeLogic` service to be registered matching that filter, it would be better if `HttpMethodFeature` didn't activate also until that is available.

Reply:
I guess a simpler way to put it is the OAuth 2 modules eating its own dogfood :)